app/src/main/java/at/roteskreuz/stopcorona/model/repositories/NotificationsRepository.kt

```diff
@@ -221,6 +237,29 @@ class NotificationsRepositoryImpl(
         ).show()
     }
 
+    override fun displayNotificationForUploadingKeysFromTheDayBefore() {
+        val title = context.string(R.string.upload_missing_keys_notification_title)
+        val message = context.string(R.string.upload_missing_keys_notification_message)
+
+        buildNotification(
+            title = title,
+            message = message,
+            pendingIntent = buildPendingIntentWithActivityStack {
+                addNextIntent(context.getDashboardActivityIntent().addFlags(firstActivityFlags))
+            },
+            channelId = NotificationChannels.CHANNEL_UPLOAD_KEYS,
+            ongoing = false
+        ).show()
+    }
+
+    override fun checkAndScheduleMissingKeysNotification(){
+        quarantineRepository.observeIfUploadOfMissingExposureKeysIsNeeded()
+            .observeOnMainThread()
```

Comment:
Please do not use Android framework in repositories if possible. Btw. Main thread here doesn't make any sense.